pkg/infrastructure/data_index.go

```diff
@@ -78,3 +89,97 @@ func InjectDataIndexURLIntoSupportingService(client *client.Client, namespace st
 	}
 	return nil
 }
+
+// MountProtoBufConfigMapsOnDeployment mounts protobuf configMaps from KogitoRuntime services into the given deployment
+func MountProtoBufConfigMapsOnDeployment(client *client.Client, deployment *appsv1.Deployment) (err error) {
+	cms, err := getProtoBufConfigMapsForAllRuntimeServices(deployment.Namespace, client)
+	if err != nil || len(cms.Items) == 0 {
+		return err
+	}
+	for _, cm := range cms.Items {
+		appendProtoBufVolumeIntoDeployment(deployment, cm)
+		appendProtoBufVolumeMountIntoDeployment(deployment, cm)
+	}
+	updateProtoBufPropInToDeploymentEnv(deployment)
+	return nil
+}
+
+// MountProtoBufConfigMapOnDataIndex mounts protobuf configMaps from KogitoRuntime services into the given deployment instance of DataIndex
+func MountProtoBufConfigMapOnDataIndex(client *client.Client, kogitoService v1alpha1.KogitoService) (err error) {
+	deployment, err := getSupportingServiceDeployment(kogitoService.GetNamespace(), client, v1alpha1.DataIndex)
+	if err != nil || deployment == nil {
+		return
+	}
+
+	cms, err := getProtoBufConfigMapsForSpecificRuntimeService(client, kogitoService.GetName(), kogitoService.GetNamespace())
+	if err != nil || len(cms.Items) == 0 {
+		return
+	}
+	for _, cm := range cms.Items {
+		appendProtoBufVolumeIntoDeployment(deployment, cm)
+		appendProtoBufVolumeMountIntoDeployment(deployment, cm)
+	}
+	updateProtoBufPropInToDeploymentEnv(deployment)
+	if err := kubernetes.ResourceC(client).Update(deployment); err != nil {
+		return err
+	}
+	return nil
+}
+
+func appendProtoBufVolumeIntoDeployment(deployment *appsv1.Deployment, cm corev1.ConfigMap) {
+	configMapDefaultMode := protoBufConfigMapVolumeDefaultMode
+	volumeExists := false
+	for _, volume := range deployment.Spec.Template.Spec.Volumes {
+		if volume.Name == cm.Name {
+			volumeExists = true
```

Comment:
You can directly call return here instead of operating with flag.

Reply:
done